File: 504/Base7.scala
object Base7 {
  def convertToBase7(num: Int): String = if (num == 0) "0" else {
    var res = StringBuilder.newBuilder
    @scala.annotation.tailrec
    def loop(num: Int): String =
      if (num == 0) res.toString().reverse
      else {
        res.append(num % 7)
        loop(num / 7)
      }

    if (num < 0) "-" + loop(-num) else loop(num)
  }

  def main(args: Array[String]): Unit = {
    val num = -7
    println(convertToBase7(num))
  }

}

File: 504/base-7.js
/**
 * @param {number} num
 * @return {string}
 */
var convertToBase7 = function (num) {
    if (!num) return "0";
    let res = [];
    let sign = num > 0 ? 1 : -1;
    num = sign * num;
    while (num != 0) {
        res.unshift(num % 7);
        num = (num - (num % 7)) / 7;
    }
    if (sign < 0) {
        res.unshift("-");
    }
    return res.join("");
};